File: this-is-coding-test/11장_모험가길드.py
N = int(input())
adventurers = list(map(int, input().split()))
adventurers.sort()

number_of_groups = 0
group = 0
for a in adventurers:
    group += 1
    if group >= a:
        number_of_groups += 1
        group = 0

print(number_of_groups)

# 5
# 2 3 1 2 2
# => 2

# 9
# 1 2 3 3 3 4 4 5 6
# => 2

# 9
# 1 1 2 2 3 4 5 5 6
# => 3

File: this-is-coding-test/11장_모험가길드.js
const N = 5;
const adventurers = [2, 3, 1, 2, 2];
adventurers.sort((a, b) => a - b);

let answer = 0;
let cnt = 0;

adventurers.forEach((v, i) => {
    cnt += 1;

    if (cnt == v) {
        answer += 1;
        cnt = 1;
    }
})

console.log(answer)
